.github/workflows/ci.yml

```diff
@@ -36,3 +35,26 @@ jobs:
       if: ${{ matrix.ghc != '8.0' }}
       run: |
          cabal test --test-show-details=streaming -j1
+
+  emulated-i386:
+    runs-on: ubuntu-latest
+    container:
+      image: i386/debian:sid
+    steps:
+    - name: install
+      run: |
+          apt-get update -y
+          apt-get install -y ghc cabal-install libghc-quickcheck2-dev libghc-tasty-dev libghc-tasty-quickcheck-dev git make curl libghc-exceptions-dev
+      shell: bash
+    - uses: actions/checkout@v1
+    - name: build
+      run: |
+         cabal --version
+         cabal update
+         cabal configure --enable-tests --constraint="quickcheck-classes -aeson -semigroupoids -vector"
```

Comment:
Try
```suggestion
         cabal configure --enable-tests --constraint="quickcheck-classes -aeson -semigroupoids -vector" --constraint "semirings -unordered-containers"
```

Reply:
i can probably remove all the constraints. Just do not want to make too many big changes.

Reply:
I think you actually want to add constraints, not remove them :) 
The goal is to build as little as possible to avoid intermittent `fdTryLock` issues.

Reply:
I was hoping to upgrade the `cabal-install` version to one where the `fdTryLock` issue is fixed.